core-parser/src/main/java/org/uniprot/core/parser/tsv/unirule/UniRuleEntryValueMapper.java

```diff
@@ -0,0 +1,194 @@
+
+    private String getAnnotationCovered(UniRuleEntry entry) {
+        List<Annotation> annotations = entry.getMainRule().getAnnotations();
+        if (Utils.notNullNotEmpty(annotations)) {
+            List<String> commentTypes =
+                    annotations.stream()
+                            .map(Annotation::getComment)
+                            .map(Comment::getCommentType)
+                            .map(CommentType::getName)
+                            .collect(Collectors.toList());
+            List<String> dbType =
+                    annotations.stream()
+                            .map(Annotation::getDbReference)
+                            .map(UniProtKBCrossReference::getDatabase)
+                            .map(UniProtKBDatabase::getName)
+                            .collect(Collectors.toList());
+            Keyword keyword =
+                    annotations.stream().map(Annotation::getKeyword).findFirst().orElse(null);
+            Gene gene = annotations.stream().map(Annotation::getGene).findFirst().orElse(null);
+            Set<String> annotCovered = new HashSet<>(commentTypes);
+            annotCovered.addAll(dbType);
+            if (Utils.notNull(keyword)) {
+                annotCovered.add("keyword");
+            }
+            if (Utils.notNull(gene)) {
+                annotCovered.add("gene");
+            }
+            return String.join(",", annotCovered);
+        }
+        return EMPTY_STRING;
+    }
+
+    private String getPredictedProteinName(UniRuleEntry uniRuleEntry) {
+        // get protein name from main rule
+        List<Annotation> annotations = uniRuleEntry.getMainRule().getAnnotations();
+        StringBuilder builder = new StringBuilder();
+        if (Utils.notNullNotEmpty(annotations)) {
+            builder = getProteinName(annotations);
+        }
+        // get protein name from case rules
+        List<CaseRule> otherRules = uniRuleEntry.getOtherRules();
+        if (Utils.notNullNotEmpty(otherRules)) {
+            for (CaseRule caseRule : otherRules) {
+                StringBuilder proteinNameBuilder = getProteinName(caseRule.getAnnotations());
+                if (builder.length() > 0) {
+                    builder.append("|");
+                }
+                if (Utils.notNullNotEmpty(proteinNameBuilder.toString())) {
+                    builder.append(proteinNameBuilder);
+                }
+            }
+        }
+
+        return builder.toString();
+    }
+
+    private StringBuilder getProteinName(List<Annotation> annotations) {
+        ProteinDescription proteinDescription =
+                annotations.stream()
+                        .map(Annotation::getProteinDescription)
+                        .findFirst()
+                        .orElse(null);
+        StringBuilder builder = new StringBuilder();
+        if (Utils.notNull(proteinDescription)) {
+            String fullName = proteinDescription.getRecommendedName().getFullName().getValue();
```

Comment:
Required: I think that in this code we might get a nullpointerException if the description do not have a recommended name. For example: this accession do not have recommended name: https://www.uniprot.org/uniprot/A0A514LIK3.txt, Can you also please add a test for this use case?

Reply:
Thanks for this finding. 👍